### PR TITLE
Update response struct, add tests

### PR DESCRIPTION
**Pull request by ChaitanyaKulkarni28**
* Accommodate the changes in response structure for `instance/gce-workload-certificates/workload-identities` and `instance/gce-workload-certificates/trust-anchors` endpoint
* Add unit tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # ignore all built binaries
-**/gce_workload_cert_refresh
-**/google_authorized_keys
-**/google_guest_agent
-**/google_metadata_script_runner
+*/gce_workload_cert_refresh
```
Comment (dorileo) [inline .gitignore:2]: Why?

Comment (dorileo) [reply .gitignore:2]: I guess the right fix would be:

```patch
+ !**/gce_workload_cert_refresh/
+ !*/google_authorized_keys/
+ !*/google_guest_agent/
+ !*/google_metadata_script_runner/
+ **/*.exe
```

Comment (ChaitanyaKulkarni28) [reply .gitignore]: Done

File: gce_workload_cert_refresh/main.go [hunk 2]
```diff
@@ -79,125 +89,62 @@ func getMetadata(ctx context.Context, key string) ([]byte, error) {
...
+		},
+        "PEER_SPIFFE_TRUST_DOMAIN_2": {
+            "trustAnchorsPem" : "<Trust bundle containing the X.509 roots certificates>",
+        }
+    }
+}
+*/
 
-	return nil
+// TrustAnchor represents a Root Certificate in metadata
```
Comment (tavishvaidya) [inline gce_workload_cert_refresh/main.go:134]: The TrustAnchorsPem can contain non-root, CA certificates as well. In this case, the certificate chain validation will stop at that certificate. Also, the TrustAnchorsPem can contain multiple certificates one after the another in no particular order. Recommend updating the comment to remove Root and clarifying that TrustAnchor can contain multiple certificates.

File: gce_workload_cert_refresh/main_test.go
Comment (tavishvaidya) [inline gce_workload_cert_refresh/main_test.go:1]: Thanks for adding these tests, they were very much needed!

**Review by tavishvaidya — APPROVED** [review]

File: gce_workload_cert_refresh/main.go (hunk 2 above)
Comment (ChaitanyaKulkarni28) [reply gce_workload_cert_refresh/main.go]: Done

**Pull request merged**